libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1094,6 +1096,18 @@ std::shared_ptr<Vote> PbftManager::generateVote(blk_hash_t const &blockhash, Pbf
   return std::make_shared<Vote>(node_sk_, std::move(vrf_sortition), blockhash);
 }
 
+uint64_t PbftManager::getThreshold(PbftVoteTypes vote_type) const {
+  switch (vote_type) {
+    case propose_vote_type:
+      return std::min<uint64_t>(NUMBER_OF_PROPOSERS, getDposTotalVotesCount());
+    case soft_vote_type:
+    case cert_vote_type:
+    case next_vote_type:
+    default:
+      return sortition_threshold_;
+  }
```

Comment:
if (vote_type == propose_vote_type) {
  return std::min<uint64_t>(NUMBER_OF_PROPOSERS, getDposTotalVotesCount());
} else {
  return sortition_threshold_;
}

I think if else will be better

Reply:
@gaoqi8649  I disagree because I think the way @MatusKysel  has it here makes it more clear that we will in the future set different thresholds for different steps.   Were that not the plan I might agree with you @gaoqi8649

Reply:
Yes that's correct @justinsnapp. This is for later, because I have noticed that algorand has committee per type, so maybe we could also introduce something like that, but so far I have no knowledge why it is better than what we have right now